### ISSUE TITLE
RLS: 1.3

### ISSUE THREAD

**Issue by simonjayhawkins** [pandas-dev/pandas#40169]
Tracking issue for the 1.3 release.

https://github.com/pandas-dev/pandas/milestone/80

Currently scheduled for May 31, 2021

List of open regressions: https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3ARegression

**Comment by simonjayhawkins** [comment]
> Currently scheduled for May 31, 2021

so we should be releasing the 1.3.0rc0 shortly (late next week or week after?), with the release of 1.3.0 a couple of weeks later

if we do 1.3.0 in mid/late June then from https://numpy.org/neps/nep-0029-deprecation_policy.html

> On Dec 26, 2021 drop support for Python 3.7 (initially released on Jun 27, 2018)

The next release can be scheduled for 6 months, and we can drop 3.7 support in the next release.

v1.2.0 was released on Dec 26, 2020, so these dates seem consistent.

when v1.3.0rc0 is released we branch straight away so that we can continue to work on master during the rc period (this is how it was done for v1.2.0rc0)

so immediately after v1.3.0rc0 we can drop support for python 3.7 on master.

---

once v1.3.0rc0 is released, we will only backport regression fixes, maybe some bugfixes, maybe extra work on new enhancements and maybe also changes to experimental features.

 ---

when v1.3.0rc0 is released, we also tag master (v2.0.0.dev0 and add a whatsnew)

so immediately after v1.3.0rc0 we can start deprecating the deprecations introduced in 1.x (may need discussion regarding deprecations introduced late in the 1.x series)

**Comment by jreback** [comment]
can u open an issue about dropping 3.7

**Comment by simonjayhawkins** [comment]
about doing it, or discuss whether to do it?

**Comment by jreback** [comment]
yeah to do it

**Comment by jreback** [comment]
@simonjayhawkins lets do the RC soon - waiting for the remaining pr labeled for 1.3 is not worth it

**Comment by simonjayhawkins** [comment]
hmm, we should get #39908 in. I think definitely worth it.

v1.2.0 was released on Dec 26, 2020 and the next release would ideally be after Dec 26, 2021 so we can drop Python 3.7

so a release around Jun 26, 2020, with a rc 2 weeks before around 12th gives us at least a few days to sort out the loose ends.

anytime next week would be consistent with these dates.

(and get 1.2.5 done first)

**Comment by simonjayhawkins** [comment]
> once v1.3.0rc0 is released, we will only backport regression fixes, maybe some bugfixes, maybe extra work on new enhancements and maybe also changes to experimental features.

I am planning to continue with some PyArrow string array changes after the rc (add arithimetic ops, min and max, more documentation and carry on with some perf improvements of the str accessor methods)

this list of open PRs milestone 1.3 https://github.com/pandas-dev/pandas/pulls?q=is%3Aopen+is%3Apr+milestone%3A1.3 contains some doc changes, which can also be done during the rc period.

#41718 is now ready and the last of the deprecations in #41485 with just a doc update outstanding https://github.com/pandas-dev/pandas/issues/41485#issuecomment-854714232 that can be done during the rc.

#41654, #41586, #39908 and #35417 have behavioral changes to core functionality that if not merged before the rc. should probably wait to the next release.

We should be on course to release 1.3.0rc0 this week assuming we can finish up the 4 PRs mentioned above.

**Comment by lithomas1** [comment]
@simonjayhawkins Will there be a 1.4 release? Looks like there is a milestone created on Github for it.

**Comment by jreback** [comment]
i created 1.4; it's possible we make this 2.0 at some point - not determined yet

**Comment by simonjayhawkins** [comment]
rc pre-release checklist

- ✅  wheels building - https://dev.azure.com/pandas-dev/pandas-wheels/_build?definitionId=3&_a=summary
- ⬜  wheels building - final check
- ⬜  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- ✅  release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.3.0.html
- ✅  ci checks https://github.com/pandas-dev/pandas/commits/master
q=is%3Apr+label%3A%22Still+Needs+Manual+Backport%22+is%3Aclosed
- ❌  no open PRs milestoned 1.3 https://github.com/pandas-dev/pandas/pulls?q=is%3Aopen+is%3Apr+milestone%3A1.3
- ⬜ no open issues milestoned 1.3 that are not suitable backport candidates https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.3

I think #41888 is the only open PR that is now outstanding

**Comment by simonjayhawkins** [comment]
> * ❌  no open PRs milestoned 1.3 https://github.com/pandas-dev/pandas/pulls?q=is%3Aopen+is%3Apr+milestone%3A1.3
> * ⬜ no open issues milestoned 1.3 that are not suitable backport candidates https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.3
> * ❌ no open blockers https://github.com/pandas-dev/pandas/milestone/80

will combine these three into checking for "Blocker for rc" label, so use that label for anything that should be done for the rc otherwise will release.

**Comment by simonjayhawkins** [comment]
rc pre-release checklist

- ⬜  MacPython.pandas nightly wheels built ok - https://dev.azure.com/pandas-dev/pandas-wheels/_build?definitionId=3&_a=summary
- ⬜  wheels building - final check
- ⬜  release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
- ⬜  release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.3.0.html
- ⬜  ci checks https://github.com/pandas-dev/pandas/commits/master
- ⬜ no open issues labelled blockers for rc https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+label%3A%22Blocker+for+rc%22+milestone%3A1.3
- ⬜ no open PRs labelled blockers for rc https://github.com/pandas-dev/pandas/pulls?q=is%3Aopen+is%3Apr+milestone%3A1.3+label%3A%22Blocker+for+rc%22

**Comment by jreback** [comment]
let's move all issues off of 1.3 except for the specifically mentioned ones. If there are no PRs then we are not waiting.

**Comment by simonjayhawkins** [comment]
I'm going though now, removing some. We need visibility of the regressions (mainly performance) for 1.3. (the regression label alone is not enough since we have unresolved regressions from earlier releases)

**Comment by simonjayhawkins** [comment]
reduced to 12 open issues. https://github.com/pandas-dev/pandas/issues?q=is%3Aopen+is%3Aissue+milestone%3A1.3

includes some regressions (behavior and performance) on master and a new bug on master.

There are a few that can maybe be closed (awaiting responses or will look in more detail if no response).

If regressions are not fixed during rc they get milestoned 1.3.1.

There are no blockers for the actual release (have removed the blocker labels that were on some)

**Comment by simonjayhawkins** [comment]
the overnight macOS py_3.7_64 failed on `test_rename_inplace`. restarted.

**Comment by simonjayhawkins** [comment]
> * ⬜  wheels building - final check

actually, I don't think this is needed in general for a rc. If the wheels fail to build and we need to make changes, we then retag, for example v1.2.1rc1.

only #41888 merged since nightly build

**Comment by simonjayhawkins** [comment]
> the overnight macOS py_3.7_64 failed on `test_rename_inplace`. restarted.

passed on restart. assume flakey.

**Comment by simonjayhawkins** [comment]
The release readiness scripts are failing on the conda test. would need to replicate locally as ci logs no help. Have not done a dummy run for this release so I will start the release (locally) and see if fixable along the way.

**Comment by simonjayhawkins** [comment]
- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/tags
- [ ] create 1.3.x branch https://github.com/pandas-dev/pandas/tree/1.3.x
- [ ] start new tag v1.4.0.dev0 https://github.com/pandas-dev/pandas/releases/tag/v1.4.0.dev0
- [ ] activate ci on 1.3.x
- [ ] open PR to build wheels https://github.com/MacPython/pandas-wheels/pull/xxx
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.3.0/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release - https://github.com/pandas-dev/pandas/releases
- [ ] upload wheels to PyPI https://pypi.org/project/pandas/
- [ ] Start v1.4.0 release notes https://github.com/pandas-dev/pandas/pull/40902
- [ ] PR to conda-forge (conda-forge bot doesn't work for rc)
- [ ] available on conda-forge https://anaconda.org/conda-forge/pandas/files
- [ ] announce

**Comment by simonjayhawkins** [comment]
xref https://github.com/pandas-dev/pandas/pull/41295#issuecomment-860037205. I'm going to do a few manual tags instead of using the scripts.

**Comment by simonjayhawkins** [comment]
> xref [#41295 (comment)](https://github.com/pandas-dev/pandas/pull/41295#issuecomment-860037205). I'm going to do a few manual tags instead of using the scripts.

I'm not sure how clean my enviroment is. so will start again

**Comment by simonjayhawkins** [comment]
@pandas-dev/pandas-core starting release now. please do not merge anything to master

- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/tags
- [ ] create 1.3.x branch https://github.com/pandas-dev/pandas/tree/1.3.x
- [ ] start new tag v1.4.0.dev0 https://github.com/pandas-dev/pandas/releases/tag/v1.4.0.dev0
- [ ] activate ci on 1.3.x
- [ ] open PR to build wheels https://github.com/MacPython/pandas-wheels/pull/xxx
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.3.0/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release - https://github.com/pandas-dev/pandas/releases
- [ ] upload wheels to PyPI https://pypi.org/project/pandas/
- [ ] Start v1.4.0 release notes https://github.com/pandas-dev/pandas/pull/40902
- [ ] PR to conda-forge (conda-forge bot doesn't work for rc)
- [ ] available on conda-forge https://anaconda.org/conda-forge/pandas/files
- [ ] announce

**Comment by simonjayhawkins** [comment]
> @pandas-dev/pandas-core starting release now. please do not merge anything to master
> 
> *

can now resume merges to master for 1.4. do not merge anything with addition to 1.3 release notes unless being backported. (i.e regressions, bugs, doc etc)

**Comment by simonjayhawkins** [comment]
I like to have the wheels downloaded before doing the github release to reduce the lag between the any notifications from Github and the wheels being available.

![image](https://user-images.githubusercontent.com/13159005/121781448-ab0c9f80-cb9c-11eb-9ba3-bb1ab1a3e51a.png)

i'm struggling with internet connectivity at the moment on 2 different networks. There's the G7 down the road so that could potentially be the reason. but with these upload speeds i'm gonna struggle getting the wheels uploaded to PyPI

The wheels have also doubled in size which is not helping either https://github.com/pandas-dev/pandas/pull/38852#issuecomment-860067753

will delay the github release while @fangchenli looks at the wheel size issue https://github.com/pandas-dev/pandas/pull/38852#issuecomment-860070198

if my connectivity is not improved by then (or I can't find another ISP with a better connectivity) then may need someone to upload the wheels to PyPI (the wheels are in the staging area)

**Comment by MarcoGorelli** [comment]
> if my connectivity is not improved by then (or I can't find another ISP with a better connectivity) then may need someone to upload the wheels to PyPI (the wheels are in the staging area)

I have until about 5:30 BST to do this if you want, do you have instructions?

**Comment by simonjayhawkins** [comment]
I would need to get the dist uploaded somewhere and we need do the Github release before uploading to PyPI

on O2 i'm getting 2Mbps which is not great, but better than BT and Three. My release machine is headless. Will see if I can tether it and still access to O2 phone.

**Comment by phofl** [comment]
I could help as well if needed. If you need help during the evening, you can ping me

**Comment by simonjayhawkins** [comment]
> I could help as well if needed. If you need help during the evening, you can ping me

Thanks @phofl I now got the wheels downloaded using a different network.

The question is now should we release with the wheel size issue? @jreback

**Comment by jreback** [comment]
can u show size vs 1.2?

yes we should release it's the rc

**Comment by simonjayhawkins** [comment]
> can u show size vs 1.2?

see https://github.com/pandas-dev/pandas/pull/38852#issuecomment-860067753
 
> yes we should release it's the rc

sure. we can always follow up with a rc1

**Comment by simonjayhawkins** [comment]
you can see the difference from 1.2.4 here https://anaconda.org/multibuild-wheels-staging/pandas/files?page=1

**Comment by jreback** [comment]
hmm are we stripping the docs?

also i am pretty sure we are generating more dtypes for some of the hashing routines so maybe unavoidable

**Comment by simonjayhawkins** [comment]
> hmm are we stripping the docs?

I think not all #38846 but we have another PR, will keep looking.

**Comment by simonjayhawkins** [comment]
> I think not all #38846 but we may have another PR, will keep looking.

https://github.com/pandas-dev/pandas-release/pull/35 was the one I was thinking of. but we ended up not removing the docs in #38846

**Comment by fangchenli** [comment]
5239a4bfa402b1dee7f1ba17abad43383320c492's wheel is below 10MB, but 8967241bc9807d7db980642716a21255102abccd's is about 20MB.

**Comment by jreback** [comment]
prob could compare the wheels and see what files are extra and then add back to gitignore

**Comment by bashtage** [comment]
It looks like the problem is that the .c source files are bring included in the wheel in 1.3.0rc0 while these are not present in the 1.2.4

From 1.3.0rc0 wheel

![image](https://user-images.githubusercontent.com/5585221/121787113-35b0c700-cbbc-11eb-834b-454d2d2fe2de.png)

**Comment by bashtage** [comment]
Size size on disk is even worse - 120MB for 1.3.0 vs about 35 for 1.2.4

**Comment by fangchenli** [comment]
Difference between 5239a4b and 8967241:
```
pandas/_libs/src/ujson/python/ujson.c
pandas/_libs/src/ujson/python/version.h
pandas/tests/io/data/excel/test3.xlsb
pandas/_libs/groupby.pyx
pandas/_libs/writers.c
pandas/tests/io/data/excel/testdateoverflow.xlsb
pandas/tests/io/data/orc/TestOrcFile.testSnappy.orc
pandas/_libs/tslibs/np_datetime.c
pandas/tests/io/data/excel/times_1904.xlsb
pandas/_libs/tslibs/tzconversion.c
pandas/tests/indexes/object/test_indexing.py
pandas/_libs/tslibs/parsing.pxd
pandas/tests/io/data/excel/test2.xlsb
pandas/tests/io/data/excel/test5.xlsb
pandas/_libs/tslibs/ccalendar.pyx
pandas/_libs/src/headers/ms_inttypes.h
pandas/tests/indexes/object/test_astype.py
pandas/_libs/tslibs/dtypes.pxd
pandas/_libs/tslibs/period.pyx
pandas/_libs/tslibs/base.c
pandas/_libs/tslibs/timezones.pyx
pandas/tests/io/data/excel/times_1900.xlsb
pandas/_libs/tslibs/timestamps.pyx
pandas/_libs/tslibs/period.pxd
pandas/_libs/khash_for_primitive_helper.pxi.in
pandas/tests/io/data/excel/blank.xlsb
pandas/_libs/src/ujson/lib/ultrajsondec.c
pandas/_libs/tslibs/ccalendar.c
pandas/_libs/tslibs/nattype.pxd
pandas/_libs/tslibs/offsets.c
pandas/_libs/src/ujson/python/objToJSON.c
pandas/_libs/reduction.c
pandas/tests/io/parser/common/test_data_list.py
pandas/_libs/ops_dispatch.pyx
pandas/_libs/src/klib/khash.h
pandas/_libs/tslibs/vectorized.c
pandas/_libs/tslibs/timestamps.c
pandas/_libs/util.pxd
pandas/_libs/writers.pyx
pandas/_libs/tslibs/timezones.c
pandas/_libs/tslibs/ccalendar.pxd
pandas/_libs/ops.c
pandas/tests/io/data/excel/blank_with_header.xlsb
pandas/_libs/internals.pyx
pandas/_libs/indexing.c
pandas/_libs/tslibs/parsing.c
pandas/tests/io/parser/common/test_read_errors.py
pandas/tests/io/data/excel/test_converters.xlsb
pandas/tests/io/parser/common/test_ints.py
pandas/_libs/index_class_helper.pxi
pandas/tests/io/data/excel/testdtype.xlsb
pandas/_libs/src/parser/io.h
pandas/tests/io/data/excel/test_squeeze.xlsb
pandas/_libs/tslibs/fields.c
pandas/_libs/parsers.c
pandas/_libs/tslibs/src/datetime/np_datetime.c
pandas/_libs/tslibs/offsets.pyx
pandas/tests/io/data/excel/test1.xlsb
pandas/_libs/sparse.pyx
pandas/_libs/tslibs/vectorized.pyx
pandas/_libs/testing.c
pandas/_libs/tslibs/np_datetime.pyx
pandas/_libs/tslibs/src/datetime/np_datetime_strings.h
pandas/_libs/join.c
pandas/_libs/tslibs/conversion.pyx
pandas/tests/io/parser/common/test_verbose.py
pandas/tests/io/parser/common/test_index.py
pandas/_libs/indexing.pyx
pandas/tests/io/data/excel/test4.xlsb
pandas/_libs/src/headers/ms_stdint.h
pandas/_libs/lib.pyx
pandas/_libs/tslibs/tzconversion.pxd
pandas/_libs/internals.c
pandas/_libs/intervaltree.pxi.in
pandas/_libs/hashing.c
pandas/_libs/index_class_helper.pxi.in
pandas/_libs/window/aggregations.cpp
pandas/_libs/properties.pyx
pandas/_libs/reshape.c
pandas/tests/io/data/excel/testskiprows.xlsb
pandas/_libs/tslibs/conversion.pxd
pandas/tests/io/data/orc/TestOrcFile.decimal.orc
pandas/tests/io/data/fixed_width/fixed_width_format.txt
pandas/_libs/src/parser/tokenizer.c
pandas/_libs/tslibs/fields.pyx
pandas/_libs/window/aggregations.pyx
pandas/_libs/tslibs/timestamps.pxd
pandas/tests/io/data/orc/TestOrcFile.test1.orc
pandas/_libs/tslibs/base.pyx
pandas/tests/io/parser/common/test_float.py
pandas/_libs/src/parse_helper.h
pandas/_libs/tslibs/util.pxd
pandas/_libs/missing.pyx
pandas/tests/io/parser/common/test_iterator.py
pandas/_libs/src/headers/stdint.h
pandas/_libs/algos_take_helper.pxi.in
pandas/_libs/reduction.pyx
pandas/_libs/hashing.pyx
pandas/_libs/tslibs/src/datetime/np_datetime_strings.c
pandas/_libs/tslibs/parsing.pyx
pandas/_libs/window/indexers.c
pandas/tests/io/data/legacy_msgpack/0.20.3/0.20.3_x86_64_darwin_3.5.2.msgpack
pandas/_libs/parsers.pyx
pandas/_libs/index.c
pandas/_libs/src/ujson/lib/ultrajson.h
pandas/_libs/tslibs/offsets.pxd
pandas/_libs/ops_dispatch.c
pandas/_libs/tslibs/nattype.c
pandas/_libs/hashtable.pyx
pandas/_libs/index.pyx
pandas/_libs/sparse_op_helper.pxi
pandas/tests/io/data/excel/test_spaces.xlsb
pandas/tests/io/data/excel/test_types.xlsb
pandas/_libs/testing.pyx
pandas/_libs/algos_common_helper.pxi.in
pandas/_libs/tslibs/timedeltas.pyx
pandas/_libs/src/klib/khash_python.h
pandas/_libs/window/indexers.pyx
pandas/_libs/src/parser/io.c
pandas/_libs/reshape.pyx
pandas/_libs/hashtable_class_helper.pxi.in
pandas/_libs/src/headers/portable.h
pandas/_libs/tslibs/strptime.c
pandas/_libs/lib.c
pandas/_libs/src/ujson/python/date_conversions.c
pandas/_libs/ops.pyx
pandas/_libs/src/skiplist.h
pandas/_libs/sparse.c
pandas/tests/io/data/excel/test_datetime_mi.xlsb
pandas/_libs/tslibs/base.pxd
pandas/tests/io/parser/common/test_common_basic.py
pandas/_libs/properties.c
pandas/_libs/src/parser/tokenizer.h
pandas/tests/io/parser/common/test_file_buffer_url.py
pandas/tests/io/data/pickle/test_py27.pkl
pandas/_libs/tslib.pyx
pandas/io/sas/sas.c
pandas/_libs/khash_for_primitive_helper.pxi
pandas/tests/io/data/excel/test_index_name_pre17.xlsb
pandas/_libs/algos_take_helper.pxi
pandas/io/sas/sas.pyx
pandas/tests/io/data/orc/TestOrcFile.emptyFile.orc
pandas/_libs/algos_common_helper.pxi
pandas/_libs/missing.pxd
pandas/_libs/join.pyx
pandas/_libs/sparse_op_helper.pxi.in
pandas/tests/io/parser/common/test_chunksize.py
pandas/_libs/src/inline_helper.h
pandas/tests/io/data/gbq_fake_job.txt
pandas/_libs/tslibs/timedeltas.pxd
pandas/tests/io/parser/common/test_inf.py
pandas/_libs/src/ujson/lib/ultrajsonenc.c
pandas/_libs/tslib.c
pandas/_libs/tslibs/conversion.c
pandas/tests/io/parser/common/test_decimal.py
pandas/_libs/hashtable.pxd
pandas/_libs/tslibs/tzconversion.pyx
pandas/_libs/hashtable_func_helper.pxi
pandas/_libs/algos.pxd
pandas/_libs/src/ujson/python/date_conversions.h
pandas/_libs/tslibs/dtypes.c
pandas/tests/io/data/pickle/test_mi_py27.pkl
pandas/_libs/lib.pxd
pandas/_libs/interval.c
pandas/_libs/groupby.c
pandas/_libs/tslibs/period.c
pandas/_libs/algos.pyx
pandas/tests/io/data/orc/TestOrcFile.testDate2038.orc
pandas/_libs/hashtable_func_helper.pxi.in
pandas/_libs/tslibs/timedeltas.c
pandas/_libs/tslibs/np_datetime.pxd
pandas/_libs/interval.pyx
pandas/_libs/src/ujson/python/JSONtoObj.c
pandas/_libs/hashtable.c
pandas/_libs/tslibs/src/datetime/np_datetime.h
pandas/_libs/tslibs/strptime.pyx
pandas/tests/io/data/orc/TestOrcFile.testDate1900.orc
pandas/tests/io/data/excel/test_multisheet.xlsb
pandas/_libs/tslibs/dtypes.pyx
pandas/_libs/missing.c
pandas/_libs/intervaltree.pxi
pandas/_libs/tslibs/timezones.pxd
pandas/_libs/khash.pxd
pandas/tests/io/data/parquet/simple.parquet
pandas/_libs/src/headers/cmath
pandas/_libs/tslibs/nattype.pyx
pandas/_libs/hashtable_class_helper.pxi
pandas/tests/io/data/excel/testmultiindex.xlsb
pandas/_libs/algos.c
```

**Comment by jreback** [comment]
ok so adding .c and .cpp and .h in the ignore section are needed

**Comment by bashtage** [comment]
What about PYX files? Not needed in the wheel either I would think.

**Comment by simonjayhawkins** [comment]
If we know what the issue is. I'll hold off releasing the current rc and do a 1.3.0rc1 tomorrow instead.

**Comment by simonjayhawkins** [comment]
I think once #41981 is merged and backported should be good for rc1

**Comment by simonjayhawkins** [comment]
@pandas-dev/pandas-core starting rc1 now. please do not merge anything to 1.3.x until tagged

- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/tags
- [ ] open PR to build wheels https://github.com/MacPython/pandas-wheels/pull/xxx
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.3.0rc1/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release - https://github.com/pandas-dev/pandas/releases
- [ ] upload wheels to PyPI https://pypi.org/project/pandas/
- [ ] PR to conda-forge (conda-forge bot doesn't work for rc)
- [ ] available on conda-forge https://anaconda.org/conda-forge/pandas/files
- [ ] announce

**Comment by simonjayhawkins** [comment]
> @pandas-dev/pandas-core starting rc1 now. please do not merge anything to 1.3.x until tagged

now tagged.

**Comment by simonjayhawkins** [comment]
will do the conda PR tomorrow.

**Comment by simonjayhawkins** [comment]
https://github.com/conda-forge/pandas-feedstock/pull/111 failing 
```
ModuleNotFoundError: No module named 'pkg_resources'
```
cc @fangchenli

**Comment by bashtage** [comment]
Is setuptools now a runtime dependency?

**Comment by simonjayhawkins** [comment]
> Is setuptools now a runtime dependency?

see #41815

**Comment by fangchenli** [comment]
#41818 only moved `setuptools` in the doc. Nothing was done to the code.

It is used in setup.py. Wasn't the package built and imported in the same virtual environment? To be more rigorous, we should check if `pkg_resources` exists in the plotting module. But it was always installed in most cases, this' why we never see this error in the CI.

@simonjayhawkins

**Comment by simonjayhawkins** [comment]
> Wasn't the package built and imported in the same virtual environment?

our conda-test passed on pandas-release, but we have `setuptools                51.0.0           py38h06a4308_2` installed in the test container.

I have a github action that doesn't use docker that started failing sometime in the last week. this is also failing for 1.2.5 release readiness so one would think unreleated. but the logs are no help. will know more once replicated that issue locally.

Maybe there are changes that could be made to pandas-feedstock, but I have little knowledge about that repo.

**Comment by bashtage** [comment]
@simonjayhawkins You could try add setuptools to the recipe to see if it works.

**Comment by simonjayhawkins** [comment]
on pandas-feedstock the builds look successful

The traceback is 
```
import: 'pandas'
Traceback (most recent call last):
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/test_tmp/run_test.py", line 2, in <module>
    import pandas
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/__init__.py", line 50, in <module>
    from pandas.core.api import (
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/core/api.py", line 48, in <module>
    from pandas.core.groupby import (
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/core/groupby/__init__.py", line 1, in <module>
    from pandas.core.groupby.generic import (
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/core/groupby/generic.py", line 73, in <module>
    from pandas.core.frame import DataFrame
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/core/frame.py", line 192, in <module>
    from pandas.core.series import Series
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/core/series.py", line 141, in <module>
    import pandas.plotting
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/plotting/__init__.py", line 58, in <module>
    from pandas.plotting._core import (
  File "/home/conda/feedstock_root/build_artifacts/pandas_1623667566353/_test_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold/lib/python3.9/site-packages/pandas/plotting/_core.py", line 10, in <module>
    import pkg_resources
ModuleNotFoundError: No module named 'pkg_resources'
```

> @simonjayhawkins You could try add setuptools to the recipe to see if it works.

maybe could add the test section. but the problem seems to be related to the imports on the plotting side. IIUC this should use the vendored code as setuptools is no longer a required dependency @fangchenli ?

**Comment by simonjayhawkins** [comment]
> IIUC this should use the vendored code as setuptools is no longer a required dependency @fangchenli ?

ahh. I think I did misunderstand. we vendor packaging not pkg_resources? so pkg_resources IS a required dependency?

**Comment by bashtage** [comment]
Should this be fixed before rc2 or should setuptools be made into a runtime dependency?  Virtually everyone has setuptools installed, but there may be some edge cases (e.g., amazon lambda) where minimal packaging is desirable.

**Comment by bashtage** [comment]
If I install pandas master in conda, and then run `conda remove setuptools --force` which skips removing packages that depend on setuptools, I see the same error.

**Comment by fangchenli** [comment]
`pkg_resources` was introduced in #41503. We could revert it for the release.

**Comment by fangchenli** [comment]
After #41678, we could redo #41503 using `importlib.metadata`. Then the dependency issue is avoided completely.

**Comment by jreback** [comment]
> After #41678, we could redo #41503 using `importlib.metadata`. Then the dependency issue is avoided completely.

for 1.3 this is not possible (but yes in the future you can do it). 

`setuptools` is a de-facto dependendy if using matplotlib (so let's just add it as an official one). i don't think its burdensome at all.

**Comment by simonjayhawkins** [comment]
To get the packages built, I could maybe add setuptools to the test section of the pandas-feedstock recipe for now.

the issue is I can't test any changes locally with `python build-locally.py`

```
+ docker run -it -v /home/simon/pandas-release/pandas-feedstock/recipe:/home/conda/recipe_root:rw,z,delegated -v /home/simon/pandas-release/pandas-feedstock:/home/conda/feedstock_root:rw,z,delegated -e CONFIG -e HOST_USER_ID -e UPLOAD_PACKAGES -e GIT_BRANCH -e UPLOAD_ON_BRANCH -e CI -e FEEDSTOCK_NAME -e CPU_COUNT -e BUILD_WITH_CONDA_DEBUG -e BUILD_OUTPUT_ID -e BINSTAR_TOKEN -e FEEDSTOCK_TOKEN -e STAGING_BINSTAR_TOKEN quay.io/condaforge/linux-anvil-comp7 /bin/bash
Traceback (most recent call last):
  File "/home/simon/pandas-release/pandas-feedstock/build-locally.py", line 96, in <module>
    main()
  File "/home/simon/pandas-release/pandas-feedstock/build-locally.py", line 90, in main
    run_docker_build(ns)
  File "/home/simon/pandas-release/pandas-feedstock/build-locally.py", line 28, in run_docker_build
    subprocess.check_call([script])
  File "/home/simon/miniconda3/envs/pandas-release/lib/python3.9/subprocess.py", line 373, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['.scripts/run_docker_build.sh']' returned non-zero exit status 139.
```

I'm on wsl2. maybe that's the issue. trying to debug but getting nowhere fast.

so I could push changes to the pandas-feedstock PR on a trial and error basis.

**Comment by simonjayhawkins** [comment]
> I'm on wsl2. maybe that's the issue. trying to debug but getting nowhere fast.

looks like a centOS thing which conda-forge uses. https://github.com/microsoft/WSL/issues/4694#issuecomment-556095344 fixes

**Comment by simonjayhawkins** [comment]
> @simonjayhawkins You could try add setuptools to the recipe to see if it works.

adding to the build, host and run sections and testing locally with python build-locally.py looks OK

will try and establish min additions (probably just run) before pushing but we will be removing this once the root issue is fixed in pandas.

**Comment by simonjayhawkins** [comment]
some green appearing on https://github.com/conda-forge/pandas-feedstock/pull/111. @jreback OK to merge that one once fully green?

**Comment by jorisvandenbossche** [comment]
> pkg_resources was introduced in #41503. We could revert it for the release.

I don't think it needs to be reverted. Can't we just move back the import inside the function?

**Comment by jreback** [comment]
> some green appearing on [conda-forge/pandas-feedstock#111](https://github.com/conda-forge/pandas-feedstock/pull/111). @jreback OK to merge that one once fully green?

ok by me, I think we should declare this dependency and NOT do #42006 . The reason is that we need this for *any* entrypoint things; sure in 3.8 can drop it i suppose (as there is a new mechanism), but up till now we have had an undeclared, but needed dependency (which just happens to be installed by matplotlib i think).

**Comment by jorisvandenbossche** [comment]
I am -1 on adding setuptools as a dependency (no problem for the RC, but IMO we should fix it for 1.3). @jreback if you want to further discuss that, please reopen #41815 (we shouldn't discuss that on this issue)

**Comment by simonjayhawkins** [comment]
> no problem for the RC, but IMO we should fix it for 1.3

indeed.

As we can get the conda packages uploaded with a tweak to the dev branch of pandas-feedstock and since the solution in rare cases where the user does not have setuptools is to just install setuptools, I think the issue does not warrant an rc2 but should be fixed before release.

Does not appear to pose an problem with clean install as setuptools gets installed.

```
(base) simon@P340:~$ conda create -n conda-test -c conda-forge -c conda-forge/label/pandas_rc pandas==1.3.0rc1
Collecting package metadata (current_repodata.json): done
Solving environment: done

## Package Plan ##

  environment location: /home/simon/miniconda3/envs/conda-test

  added / updated specs:
    - pandas==1.3.0rc1


The following NEW packages will be INSTALLED:

  _libgcc_mutex      conda-forge/linux-64::_libgcc_mutex-0.1-conda_forge
  _openmp_mutex      conda-forge/linux-64::_openmp_mutex-4.5-1_gnu
  ca-certificates    conda-forge/linux-64::ca-certificates-2021.5.30-ha878542_0
  certifi            conda-forge/linux-64::certifi-2021.5.30-py39hf3d152e_0
  ld_impl_linux-64   conda-forge/linux-64::ld_impl_linux-64-2.35.1-hea4e1c9_2
  libblas            conda-forge/linux-64::libblas-3.9.0-9_openblas
  libcblas           conda-forge/linux-64::libcblas-3.9.0-9_openblas
  libffi             conda-forge/linux-64::libffi-3.3-h58526e2_2
  libgcc-ng          conda-forge/linux-64::libgcc-ng-9.3.0-h2828fa1_19
  libgfortran-ng     conda-forge/linux-64::libgfortran-ng-9.3.0-hff62375_19
  libgfortran5       conda-forge/linux-64::libgfortran5-9.3.0-hff62375_19
  libgomp            conda-forge/linux-64::libgomp-9.3.0-h2828fa1_19
  liblapack          conda-forge/linux-64::liblapack-3.9.0-9_openblas
  libopenblas        conda-forge/linux-64::libopenblas-0.3.15-pthreads_h8fe5266_1
  libstdcxx-ng       conda-forge/linux-64::libstdcxx-ng-9.3.0-h6de172a_19
  ncurses            conda-forge/linux-64::ncurses-6.2-h58526e2_4
  numpy              conda-forge/linux-64::numpy-1.20.3-py39hdbf815f_1
  openssl            conda-forge/linux-64::openssl-1.1.1k-h7f98852_0
  pandas             conda-forge/label/pandas_rc/linux-64::pandas-1.3.0rc1-py39h83e36dc_0
  pip                conda-forge/noarch::pip-21.1.2-pyhd8ed1ab_0
  python             conda-forge/linux-64::python-3.9.4-hffdb5ce_0_cpython
  python-dateutil    conda-forge/noarch::python-dateutil-2.8.1-py_0
  python_abi         conda-forge/linux-64::python_abi-3.9-1_cp39
  pytz               conda-forge/noarch::pytz-2021.1-pyhd8ed1ab_0
  readline           conda-forge/linux-64::readline-8.1-h46c0cb4_0
  setuptools         conda-forge/linux-64::setuptools-49.6.0-py39hf3d152e_3
  six                conda-forge/noarch::six-1.16.0-pyh6c4a22f_0
  sqlite             conda-forge/linux-64::sqlite-3.35.5-h74cdb3f_0
  tk                 conda-forge/linux-64::tk-8.6.10-h21135ba_1
  tzdata             conda-forge/noarch::tzdata-2021a-he74cb21_0
  wheel              conda-forge/noarch::wheel-0.36.2-pyhd3deb0d_0
  xz                 conda-forge/linux-64::xz-5.2.5-h516909a_1
  zlib               conda-forge/linux-64::zlib-1.2.11-h516909a_1010


Proceed ([y]/n)? y

Preparing transaction: done
Verifying transaction: done
Executing transaction: done
#
# To activate this environment, use
#
#     $ conda activate conda-test
#
# To deactivate an active environment, use
#
#     $ conda deactivate

(base) simon@P340:~$ conda activate conda-test
(conda-test) simon@P340:~$ python
Python 3.9.4 | packaged by conda-forge | (default, May 10 2021, 22:13:33) 
[GCC 9.3.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import pandas as pd
>>> pd.__version__
'1.3.0rc1'
>>> 
```

**Comment by simonjayhawkins** [comment]
all wheels (including aarch 😄) and conda packages for rc1 are now available.

I will announce on `pydata@googlegroups.com` and `pandas-dev@python.org` in a few hours if no objections. 

cc @jorisvandenbossche if you want to tweet.

**Comment by simonjayhawkins** [comment]
I expect we will be releasing 1.3 sometime next week.

current status

https://github.com/pandas-dev/pandas/milestone/80

1 open PR #42068

4 open regressions #41997, #41999, #41998 and #42092 reported since rc

other open issues are mainly performance regressions.

**Comment by MarcoGorelli** [comment]
@simonjayhawkins you'd suggest on https://github.com/pandas-dev/pandas/pull/41283 to include that in 1.3 - is that still the case?

**Comment by simonjayhawkins** [comment]
yep. i'll add the milestone.

**Comment by rhshadrach** [comment]
#42286 should be an easy fix, I plan to get a PR up tonight.

**Comment by simonjayhawkins** [comment]
current status

There are several open performance issues milestoned 1.3 but non are labelled blockers.

here are some rough timings to help put these into perspective (the 2 commits compared may require more samples and we should be comparing with 1.2.5)

```
(pandas-dev) simon@T3630:~/pandas/asv_bench (master)$ asv compare v1.3.0.dev0 1.3.x --only-changed --sort ratio|head -n 20
       before           after         ratio
     [0f587028]       [f34376d1]
     <v1.3.0.dev0^0>       <1.3.x>  
+        540±30μs       41.5±0.5ms    76.86  frame_methods.Shift.time_shift(1)
+     4.80±0.08ms          180±1ms    37.54  index_object.IntervalIndexMethod.time_intersection(100000)
+      4.99±0.1ms          101±1ms    20.18  index_object.IntervalIndexMethod.time_intersection_one_duplicate(100000)
+     2.48±0.01ms       27.4±0.4ms    11.03  indexing.InsertColumns.time_assign_list_like_with_setitem
+         187±4μs      1.78±0.01ms     9.56  index_object.IntervalIndexMethod.time_intersection_one_duplicate(1000)
+     2.22±0.05μs       16.3±0.3μs     7.37  period.Indexing.time_get_loc
+     4.78±0.07ms       30.8±0.5ms     6.46  frame_methods.MaskBool.time_frame_mask_bools
+      89.2±0.4μs          459±7μs     5.14  inference.ToTimedelta.time_convert_int
+         283±3μs      1.29±0.02ms     4.54  stat_ops.FrameOps.time_op('prod', 'int', 1)
+      83.2±0.5ms          378±3ms     4.54  frame_ctor.FromDicts.time_nested_dict_int64
+      4.20±0.1μs       18.1±0.2μs     4.31  indexing.NonNumericSeriesIndexing.time_getitem_scalar('period', 'non_monotonic')
+         265±4μs      1.09±0.01ms     4.13  stat_ops.FrameOps.time_op('sum', 'int', 1)
+         518±4μs      1.84±0.01ms     3.55  stat_ops.FrameOps.time_op('mean', 'int', 1)
+         632±6μs      2.08±0.05ms     3.29  index_object.IntervalIndexMethod.time_intersection_both_duplicate(1000)
+        622±10μs      1.95±0.02ms     3.13  rolling.EWMMethods.time_ewm('Series', 1000, 'float', 'mean')
+         627±8μs      1.94±0.02ms     3.09  rolling.EWMMethods.time_ewm('Series', 10, 'float', 'mean')
+     1.44±0.06ms      4.29±0.07ms     2.97  stat_ops.FrameOps.time_op('var', 'int', 1)
```

There is an open PR to address some of these, #42293 so may have a better picture later today.

---

we have 1 issue labelled as a blocker, #41815 although we could maybe remove the blocker label since #42006 fixed the import issue.

---

behviour regressions, #42286 has an open PR to fix, #42292

---

1.3.1, https://github.com/pandas-dev/pandas/milestone/87 is scheduled for 3 weeks, July 25, 2021 so if no other blockers, any objections with moving the issues/PRs on 1.3.0 to 1.3.1 tomorrow and releasing either tomorrow/Thursday?

**Comment by jreback** [comment]
no objection

**Comment by simonjayhawkins** [comment]
pre-release checklist for v1.3.0

⬜ wheels building https://github.com/MacPython/pandas-wheels/pull/135
⬜ release readiness passing https://github.com/simonjayhawkins/pandas-release/actions?query=workflow%3A%22Tag+Release%22
⬜ release notes checked https://pandas.pydata.org/pandas-docs/dev/whatsnew/v1.3.0.html
⬜ ci checks https://github.com/pandas-dev/pandas/commits/1.3.x
⬜ no manual backports outstanding https://github.com/pandas-dev/pandas/pulls?q=is%3Apr+label%3A%22Still+Needs+Manual+Backport%22+is%3Aclosed
⬜ no open issues/PRs (except this one) https://github.com/pandas-dev/pandas/milestone/80

**Comment by simonjayhawkins** [comment]
pre-release checks passed, closed off milestone

**Comment by simonjayhawkins** [comment]
starting release now.

- [ ] tag locally
- [ ] build sdist
- [ ] pip test
- [ ] conda test
- [ ] build docs
- [ ] push tag https://github.com/pandas-dev/pandas/tags
- [ ] open PR to build wheels https://github.com/MacPython/pandas-wheels/pull/xxx
- [ ] upload docs https://pandas.pydata.org/pandas-docs/version/1.3/
- [ ] wheels in staging area https://anaconda.org/multibuild-wheels-staging/pandas/files
- [ ] github release - https://github.com/pandas-dev/pandas/releases
- [ ] upload wheels to PyPI https://pypi.org/project/pandas/
- [ ] change link for stable docs https://pandas.pydata.org/docs/
- [ ] Start v1.3.1 release notes https://github.com/pandas-dev/pandas/pull/42312
- [ ] conda-forge bot picked up on release https://github.com/conda-forge/pandas-feedstock/pull/xxx
- [ ] available on conda-forge https://anaconda.org/conda-forge/pandas/files
- [ ] announce

**Comment by simonjayhawkins** [comment]
```
HTTPError: 400 Bad Request from https://upload.pypi.org/legacy/
Project size too large. Limit for project 'pandas' total size is 10 GB. See https://pypi.org/help/#project-size-limit
```

@jreback do we need to delete some old versions?

**Comment by jreback** [comment]
wow that's crazy

**Comment by jreback** [comment]
yeah start at the oldest and just delete wheels (leave source)

**Comment by jreback** [comment]
alt reach out and see if we can get it bumped ? but may take too long to delay

**Comment by simonjayhawkins** [comment]
I don't have manage permissions for PyPI

**Comment by simonjayhawkins** [comment]
> yeah start at the oldest and just delete wheels (leave source)

or perhaps could remove the older pre-releases instead.

> alt reach out and see if we can get it bumped ? but may take too long to delay

should I do this? If you have elevated permissions it may be better if you do this.

**Comment by rhshadrach** [comment]
what about deleting pre-releases first if necessary?

**Comment by jreback** [comment]
good point
let me do

**Comment by jreback** [comment]
of course my pypi access is not working, so waiting them to reset the 2FA.

**Comment by jreback** [comment]
maybe @jorisvandenbossche can help out in the interim.

**Comment by adamjstewart** [comment]
Source for 1.3.0 isn't available on PyPI, but it sounds like you're aware of that and working on fixing the problem!

**Comment by simonjayhawkins** [comment]
> [x] available on conda-forge https://anaconda.org/conda-forge/pandas/files

except pypy. will open issue.

**Comment by jorisvandenbossche** [comment]
I am also only a "maintainer" of the pandas package on PyPI, and not an "owner". So Jeff, if you can recover your login that would be best. Or ask Wes. 

> > alt reach out and see if we can get it bumped ? but may take too long to delay
>
> should I do this? If you have elevated permissions it may be better if you do this.

Yes, we should do that. If you could open an issue for pandas requesting a larger storage limit, that would be great

**Comment by simonjayhawkins** [comment]
> Yes, we should do that. If you could open an issue for pandas requesting a larger storage limit, that would be great

from https://pypi.org/help/#project-size-limit

> first remove any unnecessary releases or individual files to lower your overall project size.

we have lots of binaries for older release, pre-releases and old windows installers

> If that is not possible, we can sometimes increase your limit

The above cleaning should free up plenty.

**Comment by jreback** [comment]
seems my 2FA is off for some reason, very odd, i filed an issue, but who knows how long this takes

**Comment by jreback** [comment]
xref https://github.com/pypa/pypi-support/issues/1192

**Comment by auvipy** [comment]
removing old pre-releases would be good move IMHO

**Comment by jreback** [comment]
@simonjayhawkins old releases have been deleted - go for uploading 1.3

**Comment by simonjayhawkins** [comment]
done.
